src/parser/mod.rs

```diff
@@ -1276,7 +1276,9 @@ impl<'a> Parser<'a> {
 
         let dialect = self.dialect;
 
-        let (next_token, next_token_index) = self.next_token_ref_with_index();
+        self.advance_token();
```

Comment:
This has two benefits: 
1. I think it is more explicit
2. It doesn't not result in a `mut` borrow of `self`(this is the key difference)